### PR TITLE
Add Quick Start API documentation and examples links

### LINKED ISSUE #92: Documentation: write Getting Started guide

### PR DESCRIPTION
**Pull request by duonoid**
# Overview
Adds "Quick Start", API documentation, and example links to README.md

## Details
* Adds "Quick Start" section, considering [opentelemetry-python](https://github.com/open-telemetry/opentelemetry-python#quick-start) and [opentelemetry-go](https://github.com/open-telemetry/opentelemetry-go#quick-start) examples as precedent
* NOTE: python example has `ConsoleSpanExporter` available
* TODO: API documentation and `examples` links are stubbed (no content) at this time
* Tested via copy/paste into local ruby script
* Adds ConsoleSpanExporter (similar to the one in #131)

## Fixes
* Fixes #92

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,6 +43,37 @@ The services provided are:
...
+```ruby
+require 'opentelemetry/sdk'
+
+# Set preferred tracer implementation:
+SDK = OpenTelemetry::SDK
+
+factory = OpenTelemetry.tracer_factory = SDK::Trace::TracerFactory.new
+factory.add_span_processor(
+  SDK::Trace::Export::SimpleSpanProcessor.new(
+    # TODO: opentelemetry-python uses ConsoleSpanExporter:
```
Comment (mwear) [inline README.md]: I would be ok moving the ConsoleSpanExporter from #131 into the SDK. What do you think @fbogsany?

File: README.md [hunk 2]
```diff
@@ -43,6 +43,37 @@ The services provided are:
...
 
+## Quick Start
+
+```ruby
+require 'opentelemetry/sdk'
+
+# Set preferred tracer implementation:
+SDK = OpenTelemetry::SDK
+
+factory = OpenTelemetry.tracer_factory = SDK::Trace::TracerFactory.new
```
Comment (mwear) [inline README.md]: In the long haul, I think we are definitely in need of a `OpenTelemetry::SDK.configure` block to help users get everything setup (and possibly validate that they've setup everything up properly). For now, this is the state of the world.

File: README.md [hunk 3]
```diff
@@ -43,6 +43,37 @@ The services provided are:
...
+factory.add_span_processor(
+  SDK::Trace::Export::SimpleSpanProcessor.new(
+    # TODO: opentelemetry-python uses ConsoleSpanExporter:
+    SDK::Trace::Export::InMemorySpanExporter.new
+  )
+)
+
+tracer = factory.tracer('trace_name', _version = 'semver:1.0')
```
Comment (mwear) [inline README.md]: `version` is a positional argument on `TracerFactory#tracer`. I'm uncertain what the tracer name should be for user _application code_, but we should probably use that here. My guess is that it should be the application name, but I'd be open to any other opinions. If we go with that, maybe `trace_name` could be some fictitious application name, such as `my_app`, `my_webstore`, or something slightly more imaginative.

File: README.md (hunk 2 above)
Comment (fbogsany) [reply README.md]: Agreed - this hurts every time I read it 😞

File: README.md (hunk 3 above)
Comment (fbogsany) [reply README.md]: From the [OTEP](https://github.com/open-telemetry/oteps/blob/master/text/0016-named-tracers.md):
> If a library (or application) has instrumentation built-in, it is both the instrumenting and instrumented library and should pass its own name here

So yeah, application name appears to be correct.

File: README.md (hunk 1 above)
Comment (fbogsany) [reply README.md]: WFM 👍

Comment (duonoid) [reply README.md]: @mwear I've added a ConsoleSpanExporter, for review.

File: README.md (hunk 3 above)
Comment (duonoid) [reply README.md]: I've updated to use the app or gem name and version (from `Gem.loaded_specs[]`) when creating a tracer.

**Comment by duonoid** [comment]
@mwear I've addressed most of the action points on this PR, ready for another look (thanks!).

File: sdk/lib/opentelemetry/sdk/trace/export/console_span_exporter.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+module OpenTelemetry
+  module SDK
+    module Trace
+      module Export
+        # Outputs {SpanData} to the console.
+        #
+        # Potentially useful for exploratory purposes.
+        class ConsoleSpanExporter
+          ResultCodes = OpenTelemetry::SDK::Trace::Export
```
Comment (fbogsany) [inline sdk/lib/opentelemetry/sdk/trace/export/console_span_exporter.rb]: Maybe make this a `private_constant`?

File: sdk/test/opentelemetry/sdk/trace/export/console_span_exporter_test.rb [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  end
+
+  it 'accepts an Array of Spans as arg to #export and succeeds' do
+    _(exporter.export(spans)).must_equal export::SUCCESS
+  end
+
+  it 'accepts an Enumerable of Spans as arg to #export and succeeds' do
+    enumerable = Struct.new(:span0, :span1).new
+    enumerable.span0 = spans[0]
+    enumerable.span1 = spans[1]
```
Comment (fbogsany) [inline sdk/test/opentelemetry/sdk/trace/export/console_span_exporter_test.rb]: Can just do this in one line?
```ruby
enumerable = Struct.new(:span0, :span1).new(span[0], span[1])
```

File: sdk/lib/opentelemetry/sdk/trace/export/console_span_exporter.rb (hunk 4 above)
Comment (duonoid) [reply sdk/lib/opentelemetry/sdk/trace/export/console_span_exporter.rb]: :heavy_check_mark: thanks!

File: sdk/test/opentelemetry/sdk/trace/export/console_span_exporter_test.rb (hunk 5 above)
Comment (duonoid) [reply sdk/test/opentelemetry/sdk/trace/export/console_span_exporter_test.rb]: I've refactored this file and nearby tests to use your suggestions.  thanks! :heavy_check_mark:

**Comment by duonoid** [comment]
@fbogsany I've addressed the last round of change requests... thanks for the feedback!

**Comment by mwear** [comment]
@duonoid I would like to merge this, but for some reason the CLA check is hanging. I don't have a magic button to press. If it stays in the this state, perhaps you can `git pull`, `git commit --amend` (just to get a new sha), then `git push -f`. There might be an easier way, so if you have ideas, go ahead and try them.

**Pull request merged**